frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestTable.tsx

```diff
@@ -0,0 +1,313 @@
+        requiredApprovals: 1,
+    });
+
+    const theme = useTheme();
+    const projectId = useRequiredPathParam('projectId');
+    const { data, loading, refetchChangeRequestConfig } =
+        useChangeRequestConfig(projectId);
+    const { updateChangeRequestEnvironmentConfig } = useChangeRequestApi();
+    const { setToastData, setToastApiError } = useToast();
+
+    const onRowChange =
+        (
+            enableEnvironment: string,
+            isEnabled: boolean,
+            requiredApprovals: number
+        ) =>
+        () => {
+            setDialogState({
+                isOpen: true,
+                enableEnvironment,
+                isEnabled,
+                requiredApprovals,
+            });
+        };
+
+    const onConfirm = async () => {
+        if (dialogState.enableEnvironment) {
+            await updateConfiguration();
+        }
+        setDialogState(state => ({ ...state, isOpen: false }));
+    };
+
+    async function updateConfiguration(config?: IChangeRequestConfig) {
+        try {
+            await updateChangeRequestEnvironmentConfig(
+                config || {
+                    project: projectId,
+                    environment: dialogState.enableEnvironment,
+                    enabled: !dialogState.isEnabled,
+                    requiredApprovals: dialogState.requiredApprovals,
+                }
+            );
+            setToastData({
+                type: 'success',
+                title: 'Updated change request status',
+                text: 'Successfully updated change request status.',
+            });
+            await refetchChangeRequestConfig();
+        } catch (error) {
+            setToastApiError(formatUnknownError(error));
+        }
+    }
+
+    const approvalOptions = Array.from(Array(10).keys())
+        .map(key => String(key + 1))
+        .map(key => {
+            const labelText = key === '1' ? 'approval' : 'approvals';
+            return {
+                key,
+                label: `${key} ${labelText}`,
+                sx: { 'font-size': theme.fontSizes.smallBody },
+            };
+        });
+
+    function onRequiredApprovalsChange(original: any, approvals: string) {
+        updateConfiguration({
+            project: projectId,
+            environment: original.environment,
+            enabled: original.changeRequestEnabled,
+            requiredApprovals: Number(approvals),
+        });
+    }
+
+    const columns = useMemo(
+        () => [
+            {
+                Header: 'Environment',
+                accessor: 'environment',
+                disableSortBy: true,
+            },
+            {
+                Header: 'Type',
+                accessor: 'type',
+                disableGlobalFilter: true,
+                disableSortBy: true,
+            },
+            {
+                Header: 'Required approvals',
+                align: 'center',
+                Cell: ({ row: { original } }: any) => {
```

Comment:
maybe this can be extracted to a component

Reply:
Agreed, just moved the implementation for the moment, but it could use some cleanup